pkgs/games/humblebundle/spacechem.nix

```diff
@@ -0,0 +1,69 @@
+{ stdenv, fetchHumbleBundle, unzip, makeWrapper,
+SDL, SDL_mixer, SDL_image, SDL_net, SDL_gfx, SDL_ttf,
+smpeg, zlib, mono, monoDLLFixer, sqlite }:
+
+let
+  version = "1.0.12-2";
+  versionName = "15005";
+  arch = {
+    "i686-linux" = "i686";
+    "x86_64-linux" = "x86_64";
+  }.${stdenv.system};
+in stdenv.mkDerivation rec {
+  name = "spacechem-${version}";
+
+  src = fetchHumbleBundle {
+    name = "spacechem-linux-1345144627-amd64.deb";
+    machineName = "spacechem_android_pc_soundtrack_linux";
+    downloadName = "amd64.deb";
+    #md5 = "2df1aa8d2709c94e7990742210c35940";
+    md5 = "c08fb05e1909c0b4ccae82a862da6e3a";
+  };
+
+  phases = [ "installPhase" "fixupPhase" ];
+
+  buildInputs = [ makeWrapper ];
+
+  inherit monoDLLFixer;
+
+  installPhase = let
+    rpath = stdenv.lib.makeLibraryPath [ SDL SDL_mixer zlib stdenv.cc.cc ];
+    dllmap = {
+      SDL = "${SDL}/lib/libSDL.so";
+      SDL_image = "${SDL_image}/lib/libSDL_image.so";
+      SDL_mixer = "${SDL_mixer}/lib/libSDL_mixer.so";
+      SDL_ttf = "${SDL_ttf}/lib/libSDL_ttf.so";
+      SDL_net = "${SDL_net}/lib/libSDL_net.so";
+      smpeg = "${smpeg}/lib/libsmpeg.so";
+      SDL_gfx = "${SDL_gfx}/lib/libSDL_gfx.so";
+    };
+  in ''
+    set -x
+    mkdir $out
+    ar p $src data.tar.gz | (cd $out; tar xvzf - ./opt)
```

Comment:
Maybe something like `unpackCmd = "ar p \"$curSrc\" data.tar.gz | tar xz opt";`?

Reply:
Well, this unpacks directly to $out, avoiding an extra move/copy step. is there a way to mimic that with unpackCmd ?

Reply:
You could also specify `-C "$out"` then.

As for the size issue... I'm quite aware that the whole `unpack -> build -> install` is something that really could be improved. I don't know about Spacechem, but something like eg. Pyre has > 6 GB to unpack.

So the long-term solution would probably be something like unzipping the game's package directly into a store path (as a fixed-output derivation) and reference them from within the `buildGame` derivation. However, this could require more patching and would mess with the directory structure (also it wouldn't work so well with the md5 hashes for HumbleBundle, but I guess it's time to simply deprecate them anyway, regardless of whether HB will move on or not). Another idea I had is to use the sandbox to overlay paths directly via bind mounts into the chroot.